test/models/concerns/host_managed_extensions_test.rb

```diff
@@ -176,4 +176,81 @@ def test_search_known_traces
       assert_includes ::Host::Managed.search_for("trace_helper = \"sudo systemctl restart sshd\""), @foreman_host
     end
   end
+
+  class HostManagedExtensionsKickstartTest < ActiveSupport::TestCase
+    def setup
+      disable_orchestration # disable foreman orchestration
+      @distro = katello_repositories(:fedora_17_x86_64)
+      @os = ::Redhat.create_operating_system('RedHat', '17', '0')
+      @os.stubs(:kickstart_repos).returns([@distro])
+      @arch = architectures(:x86_64)
+      @distro_cv = @distro.content_view
+      @distro_env = @distro.environment
+      @content_source = FactoryBot.create(:smart_proxy,
+                                          name: "foobar",
+                                          url: "http://example.com/",
+                                          lifecycle_environments: [@distro_env])
+      @medium = FactoryBot.create(:medium, operatingsystems: [@os])
+
+      @host = FactoryBot.create(:host, operatingsystem: @os, arch: @arch)
+      Support::HostSupport.attach_content_facet(@host, @distro_cv, @distro_env)
+      @host.content_facet.content_source = @content_source
+      @host.save!
+    end
+
+    def test_set_medium
+      @host.medium = @medium
+      assert_valid @host
+      assert_equal @host.medium, @medium
+    end
+
+    def test_set_installation_medium
+      @host.content_facet.kickstart_repository = @distro
+      assert_valid @host
+      assert_equal @host.content_facet.kickstart_repository, @distro
+    end
+
+    def test_change_medium_to_kickstart_repository
+      @host.medium = @medium
+      assert @host.save
```

Comment:
Do we need `save`, or can we just call `valid`?

Reply:
It needs saving after setting the first value, so the second value change causes Rails to have the right magic `_changed?` values.